server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -1157,6 +1157,14 @@ public synchronized boolean updateIndexMetadata(IndexMetadata indexMetadata) {
             // nothing to update, same settings
             return false;
         }
+
+        // The index.mode setting also validates other index settings,
+        // so we need to read the new settings to ensure index.mode validation doesn't fail.
+        // Also in the case the index.mode hasn't been specified in newSettings (and default applies)
+        // (Note when creating a new index this happens when we read the 'mode' field in the constructor of this class
+        // regardless of whether the index.mode setting has been specified)
+        IndexSettings.MODE.get(newSettings);
```

Comment:
Parsing the index.mode setting with new settings triggers the validation. We don't need to keep the result around.

I chose this place to do this because in the case when index.time_series.end_time is defined at index creation time and the index.mode setting isn't defined then the parsing of the index.mode setting in the constructor of this class triggers that validation that fails creating the index. So the logic used at index creation time and setting update time is at least in the same class.

Reply:
Can we do it somewhere here https://github.com/elastic/elasticsearch/blob/dea593db3f4d7d40ed23c0574a6a3f3fb8b3470e/server/src/main/java/org/elasticsearch/index/IndexSettings.java#L620?

Reply:
Maybe yes if we add the `MODE` to `settings()` below as we need the `inde.mode` to be available in the settings to validate it against?

Reply:
This seems to work out. I like this approach better as the logic is contained with in the `TIME_SERIES_END_TIME` setting anonymous inner class.